drivers/modem/modem_context.h

```diff
@@ -83,6 +85,10 @@ struct modem_context {
 	/* command handler config */
 	struct modem_cmd_handler cmd_handler;
 
+	/* SMS functions */
+	send_sms_func send_sms;
+	recv_sms_func recv_sms;
```

Comment:
The naming of these function pointers in other part of the system seems to be `xxxx_func_t` so I propose we follow the same naming convention.